lib/cn/cn_tree.c

```diff
@@ -1436,7 +1445,20 @@ cn_comp_update_spill(struct cn_compaction_work *work, struct kvset **kvsets)
     INIT_LIST_HEAD(&retired_kvsets);
 
     rmlock_wlock(&tree->ct_lock);
-    {
+
+    do {
+
+        /* If a node split occurred in the middle of a spill then abort and retry.
+         *
+         * TODO: Once incremental root spill is available we should be able to
+         * track splits on a per-leaf basis to avoid aborting disjoint spills.
+         */
+        if (work->cw_splitgen != tree->ct_splitgen) {
+            work->cw_err = merr(EAGAIN);
+            ev_debug(1);
+            break;
```

Comment:
This function cannot cause failures.  The operation is committed.  If you want to leave it temporarily, make it an abort.

Reply:
That makes sense, I didn't like the fact that commit could fail.  So basically cn_comp_update_split() needs some serious work so that it cannot fail.
Not sure what you mean by "leave it temporarily, make it an abort" ????